Jackal/Jackal Shared/Playing Field/Shared/FieldNodes/ShipNode.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ShipNode.swift
+//  Jackal iOS
+//
+//  Created by Alexander Savonin on 22/03/2018.
+//  Copyright © 2018 Boring Company. All rights reserved.
+//
+
+struct ShipNode: FieldNodeDescribing {
+    
+    // MARK: Public Data Structures
+    enum ShipType: String {
+        case blackShip
+        case greenShip
+        case purpleShip
+        case redShip
+    }
+    
+    // MARK: Public Properties
+    var moveType: MoveType
+    var rotation: Rotation
+    var canContainObject = true
+    var canStay = true
+    var actionType: ActionType = .permanent
+    var textureName: String
+    
+    // MARK: Lifecycle
+    init(type: ShipType, rotation: Rotation) {
+        let moves: [Move] = [Move(x:  0, y: +1, rotation: rotation),
```

Comment:
What about dynamic rotation applying? E.X. 
var moveType: MoveType {
get {
switch value {
.oneOf(moves):
return .oneOf(moves.foreach(applyRotation))
}
}
}


I'm not sure about idea to store rotation inside move, for me, rotation belongs to node only

Reply:
yeah, your idea sounds better to me.

Reply:
I agree, that there's no need to apply rotation on initialization phase.
Just for note: Don't use explicit `get` when there is no setter